### PR TITLE
community: Overhaul Databricks provider documentation

### PR DESCRIPTION
**Pull request by B-Step62**
**Description**: Update [Databricks Provider](https://python.langchain.com/v0.2/docs/integrations/providers/databricks/) documentations to the latest component notebooks and draw better navigation path to related notebooks.

### CODE REVIEW

**Comment by B-Step62** [comment]
@isahers1 Vercel deployment has failed, but I can't access to the details/inspect due to 404. Locally running `make doc_build` passes so I have not idea what caused the deployment issue. Would you mind help me with debugging the issue?

File: docs/docs/integrations/providers/databricks.md [hunk 1]
```diff
@@ -1,87 +1,111 @@
...
+1. 🚀 **Model Serving** - Access state-of-the-art LLMs or your fine-tuned models on [Databricks Model Serving](https://www.databricks.com/product/model-serving), via a highly available and low-latency inference endpoint. LangChain provides LLM (`Databricks`), Chat Model (`ChatDatabricks`), and Embeddings (`DatabricksEmbeddings`) implementations, streamlines the integration of your models hosted on Databricks Model Serving into your LangChain application.
+2. 📃 **Vector Search** - [Databricks Vector Search](https://www.databricks.com/product/machine-learning/vector-search) is a serverless vector database seamlessly integrated in teh Databricks Platform. Using `DatabricksVectorSearch`, you can incorporate the highly scalable and reliable similarity search engine into your LangChain application.
+3. 📊 **MLflow** - [MLflow](https://mlflow.org/) is an open-source platrofm to manage the MLf lifecycle, including experiment management, evaluation, tracing, deployment, and more. [MLflow LangChain Integration](../providers/mlflow_tracking.ipynb) streamline the process of developing and operating modern compound ML systems.
```
Comment (ccurme) [inline docs/docs/integrations/providers/databricks.md:10]:
```suggestion
3. 📊 **MLflow** - [MLflow](https://mlflow.org/) is an open-source platrofm to manage the MLf lifecycle, including experiment management, evaluation, tracing, deployment, and more. [MLflow LangChain Integration](/docs/integrations/providers/mlflow_tracking) streamline the process of developing and operating modern compound ML systems.
```

**Review by ccurme — COMMENTED** [review]
Build is failing due to broken links
```
[ERROR] Error: Docusaurus found broken links!
```
Could you update some of the documentation links? I've included a suggested change for one of them.

Thank you!

File: docs/docs/integrations/providers/databricks.md (hunk 1 above)
Comment (B-Step62) [reply docs/docs/integrations/providers/databricks.md:10]: Ahh makes sense, thank you so much for the pointers @ccurme !!

File: docs/docs/integrations/providers/databricks.md [hunk 2]
```diff
@@ -1,87 +1,111 @@
...
 
 Databricks embraces the LangChain ecosystem in various ways:
 
-1. Databricks connector for the SQLDatabase Chain: SQLDatabase.from_databricks() provides an easy way to query your data on Databricks through LangChain
-2. Databricks MLflow integrates with LangChain: Tracking and serving LangChain applications with fewer steps
-3. Databricks as an LLM provider: Deploy your fine-tuned LLMs on Databricks via serving endpoints or cluster driver proxy apps, and query it as langchain.llms.Databricks
-4. Databricks Dolly: Databricks open-sourced Dolly which allows for commercial use, and can be accessed through the Hugging Face Hub
+1. 🚀 **Model Serving** - Access state-of-the-art LLMs or your fine-tuned models on [Databricks Model Serving](https://www.databricks.com/product/model-serving), via a highly available and low-latency inference endpoint. LangChain provides LLM (`Databricks`), Chat Model (`ChatDatabricks`), and Embeddings (`DatabricksEmbeddings`) implementations, streamlines the integration of your models hosted on Databricks Model Serving into your LangChain application.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:8]:
```suggestion
1. 🚀 **Model Serving** - Access state-of-the-art LLMs or your fine-tuned models on [Databricks Model Serving](https://www.databricks.com/product/model-serving), via a highly available and low-latency inference endpoint. LangChain provides LLM (`Databricks`), Chat Model (`ChatDatabricks`), and Embeddings (`DatabricksEmbeddings`) implementations, streamlining the integration of your models hosted on Databricks Model Serving with your LangChain applications.
```

File: docs/docs/integrations/providers/databricks.md [hunk 3]
```diff
@@ -1,87 +1,111 @@
...
+2. 📃 **Vector Search** - [Databricks Vector Search](https://www.databricks.com/product/machine-learning/vector-search) is a serverless vector database seamlessly integrated in teh Databricks Platform. Using `DatabricksVectorSearch`, you can incorporate the highly scalable and reliable similarity search engine into your LangChain application.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:9]:
```suggestion
2. 📃 **Vector Search** - [Databricks Vector Search](https://www.databricks.com/product/machine-learning/vector-search) is a serverless vector database seamlessly integrated within the Databricks Platform. Using `DatabricksVectorSearch`, you can incorporate the highly scalable and reliable similarity search engine into your LangChain applications.
```

File: docs/docs/integrations/providers/databricks.md [hunk 4]
```diff
@@ -1,87 +1,111 @@
...
+3. 📊 **MLflow** - [MLflow](https://mlflow.org/) is an open-source platrofm to manage the MLf lifecycle, including experiment management, evaluation, tracing, deployment, and more. [MLflow LangChain Integration](/docs/integrations/providers/mlflow_tracking) streamline the process of developing and operating modern compound ML systems.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:10]:
```suggestion
3. 📊 **MLflow** - [MLflow](https://mlflow.org/) is an open-source platform to manage the full ML lifecycle, including experiment management, evaluation, tracing, deployment and more. [MLflow's LangChain Integration](/docs/integrations/providers/mlflow_tracking) streamlines the process of developing and operating modern compound ML systems.
```

File: docs/docs/integrations/providers/databricks.md [hunk 5]
```diff
@@ -1,87 +1,111 @@
...
+4. 🌐 **SQL Database** - [Databricks SQL](https://www.databricks.com/product/databricks-sql) is integrated with `SQLDatabase` in LangChain, allowing you to access the auto-optimizing best performant data warehouse.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:11]:
```suggestion
4. 🌐 **SQL Database** - [Databricks SQL](https://www.databricks.com/product/databricks-sql) is integrated with `SQLDatabase` in LangChain, allowing you to access the auto-optimizing, exceptionally performant data warehouse.
```
just to avoid flamewars ;)

File: docs/docs/integrations/providers/databricks.md [hunk 6]
```diff
@@ -1,87 +1,111 @@
...
+5. 💡 **Open Models** - Databricks open-sourced [DBRX](https://www.databricks.com/blog/introducing-dbrx-new-state-art-open-llm) model which allows for commercial use, and can be accessed through the [Hugging Face Hub](https://huggingface.co/databricks/dbrx-instruct).
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:12]: Since we're hosting a fleet of these now, should we rephrase this to make a list of models that can be used (non-exhaustive) that highlights DBRX along with other popular open-source transformers-compatible models?

File: docs/docs/integrations/providers/databricks.md [hunk 7]
```diff
@@ -1,87 +1,111 @@
...
 
-Databricks connector for the SQLDatabase Chain
-----------------------------------------------
-You can connect to [Databricks runtimes](https://docs.databricks.com/runtime/index.html) and [Databricks SQL](https://www.databricks.com/product/databricks-sql) using the SQLDatabase wrapper of LangChain. 
+Chat Model
+----------
 
+`ChatDatabricks` is a Chat Model class to access chat endpoints hosted on Databricks, including state-of-the-art models such as Llama3, Mixtral, and DBRX, as well as your fine-tuned models.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:17]:
```suggestion
`ChatDatabricks` is a Chat Model class to access chat endpoints hosted on Databricks, including state-of-the-art models such as Llama3, Mixtral, and DBRX, as well as your own fine-tuned models.
```

File: docs/docs/integrations/providers/databricks.md [hunk 8]
```diff
@@ -1,87 +1,111 @@
...
 
-Databricks MLflow integrates with LangChain
--------------------------------------------
+```
+from langchain_community.chat_models.databricks import ChatDatabricks
+
+chat_model = ChatDatabricks(endpoint="databricks-meta-llama-3-70b-instruct")
+```
+
+See [usage example](/docs/integrations/chat/databricks) for more guidance for how to use it within your LangChain application.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:25]:
```suggestion
See the [usage example](/docs/integrations/chat/databricks) for more guidance on how to use it within your LangChain application.
```

File: docs/docs/integrations/providers/databricks.md [hunk 9]
```diff
@@ -1,87 +1,111 @@
...
+
+```
+from langchain_community.llm.databricks import Databricks
 
-MLflow is an open-source platform to manage the ML lifecycle, including experimentation, reproducibility, deployment, and a central model registry. See the notebook [MLflow Callback Handler](/docs/integrations/providers/mlflow_tracking) for details about MLflow's integration with LangChain.
+llm = Databricks(endpoint="your-completion-endpoint")
+```
 
-Databricks provides a fully managed and hosted version of MLflow integrated with enterprise security features, high availability, and other Databricks workspace features such as experiment and run management and notebook revision capture. MLflow on Databricks offers an integrated experience for tracking and securing machine learning model training runs and running machine learning projects. See [MLflow guide](https://docs.databricks.com/mlflow/index.html) for more details.
+See [usage example](/docs/integrations/llms/databricks) for more guidance for how to use it within your LangChain application.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:38]:
```suggestion
See the [usage example](/docs/integrations/llms/databricks) for more guidance on how to use it within your LangChain application.
```

File: docs/docs/integrations/providers/databricks.md [hunk 10]
```diff
@@ -1,87 +1,111 @@
...
 
-Databricks MLflow makes it more convenient to develop LangChain applications on Databricks. For MLflow tracking, you don't need to set the tracking uri. For MLflow Model Serving, you can save LangChain Chains in the MLflow langchain flavor, and then register and serve the Chain with a few clicks on Databricks, with credentials securely managed by MLflow Model Serving.
 
-Databricks External Models
---------------------------
+Embeddings
+----------
 
-[Databricks External Models](https://docs.databricks.com/generative-ai/external-models/index.html) is a service that is designed to streamline the usage and management of various large language model (LLM) providers, such as OpenAI and Anthropic, within an organization. It offers a high-level interface that simplifies the interaction with these services by providing a unified endpoint to handle specific LLM related requests. The following example creates an endpoint that serves OpenAI's GPT-4 model and generates a chat response from it:
+`DatabricksEmbeddings` is an Embeddings class to access text-embedding endpoints hosted on Databricks, including state-of-the-art models such as BGE, as well as your fine-tuned models.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:44]:
```suggestion
`DatabricksEmbeddings` is an Embeddings class to access text-embedding endpoints hosted on Databricks, including state-of-the-art models such as BGE, as well as your own fine-tuned models.
```

File: docs/docs/integrations/providers/databricks.md [hunk 11]
```diff
@@ -1,87 +1,111 @@
...
-from langchain_community.chat_models import ChatDatabricks
-from langchain_core.messages import HumanMessage
-from mlflow.deployments import get_deploy_client
+```
+from langchain_community.embeddings import DatabricksEmbeddings
+
+embeddings = DatabricksEmbeddings(endpoint="databricks-bge-large-en")
+```
 
+See [usage example](/docs/integrations/text_embedding/databricks) for more guidance for how to use it within your LangChain application.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:52]:
```suggestion
See the [usage example](/docs/integrations/text_embedding/databricks) for more guidance on how to use it within your LangChain application.
```

File: docs/docs/integrations/providers/databricks.md [hunk 12]
```diff
@@ -1,87 +1,111 @@
...
-                    },
-                },
-            }
-        ],
-    },
+
+Vector Search
+-------------
+
+Databricks Vector Search is a serverless similarity search engine that allows you to store a vector representation of your data, including metadata, in a vector database. With Vector Search, you can create auto-updating vector search indexes from Delta tables managed by Unity Catalog and query them with a simple API to return the most similar vectors.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:58]: Can we add links for Delta and UC?

File: docs/docs/integrations/providers/databricks.md [hunk 13]
```diff
@@ -1,87 +1,111 @@
...
 )
-chat = ChatDatabricks(endpoint=name, temperature=0.1)
-print(chat([HumanMessage(content="hello")]))
-# -> content='Hello! How can I assist you today?'
+docs = dvs.similarity_search("What is vector search?)
 ```
 
-Databricks Foundation Model APIs
---------------------------------
+See [usage example](/docs/integrations/vectorstores/databricks_vector_search) for how to set up vector indices and integrate it with LangChain.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:69]:
```suggestion
See the [usage example](/docs/integrations/vectorstores/databricks_vector_search) for how to set up vector indices and integrate them with LangChain.
```

File: docs/docs/integrations/providers/databricks.md [hunk 14]
```diff
@@ -1,87 +1,111 @@
...
-from langchain_community.embeddings import DatabricksEmbeddings
+MLflow Integration
+------------------
 
+In the context of LangChain integration, MLflow provides the following capabilities:
 
-embeddings = DatabricksEmbeddings(endpoint="databricks-bge-large-en")
-print(embeddings.embed_query("hello")[:3])
-# -> [0.051055908203125, 0.007221221923828125, 0.003879547119140625, ...]
+- **Experiment Tracking**: Tracks and stores models and artifacts from your LangChain experiments.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:77]:
```suggestion
- **Experiment Tracking**: Tracks and stores model, artifacts, and traces from your LangChain experiments.
```

File: docs/docs/integrations/providers/databricks.md [hunk 15]
```diff
@@ -1,87 +1,111 @@
...
+- **Dependency Management**: Automatically records dependency libraries, ensuring consistency between development and production environments.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:78]:
```suggestion
- **Dependency Management**: Automatically records dependency libraries, ensuring consistency among  development, staging and production environments.
```

File: docs/docs/integrations/providers/databricks.md [hunk 16]
```diff
@@ -1,87 +1,111 @@
...
+- **Model Evaluation** Offers native capabilities for evaluating LangChain applications.
+- **Tracing**: Visually traces data flows through your LangChain application.
+
+See [MLflow LangChain Integration](/docs/integrations/providers/mlflow_tracking) to learn about the full ability of LangChain with MLflow and code examples.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:82]:
```suggestion
See [MLflow LangChain Integration](/docs/integrations/providers/mlflow_tracking) to learn about the full capabilities of using MLflow with LangChain through extensive code examples and guides.
```

File: docs/docs/integrations/providers/databricks.md [hunk 17]
```diff
@@ -1,87 +1,111 @@
...
-The notebook [Wrap Databricks endpoints as LLMs](/docs/integrations/llms/databricks#wrapping-a-serving-endpoint-custom-model) demonstrates how to serve a custom model that has been registered by MLflow as a Databricks endpoint.
-It supports two types of endpoints: the serving endpoint, which is recommended for both production and development, and the cluster driver proxy app, which is recommended for interactive development. 
+See [Databricks SQL Agent](https://docs.databricks.com/en/large-language-models/langchain.html#databricks-sql-agent) for how to connect Databricks SQL with your LangChain Agent as a powerful querying tool.
 
+Open Models
+-----------
 
-Databricks Vector Search
-------------------------
+To directly integrate Databricks's open models hosted on HuggingFace, you can use [HuggingFace Integration](/docs/integrations/platforms/huggingface) of LangChain.
```
Comment (BenWilson2) [inline docs/docs/integrations/providers/databricks.md:98]:
```suggestion
To directly integrate Databricks's open models hosted on HuggingFace, you can use the [HuggingFace Integration](/docs/integrations/platforms/huggingface) of LangChain.
```

**Review by BenWilson2 — APPROVED** [review]
Tiny nits; otherwise LGTM!
